server/routerlicious/packages/services/src/containerManager.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ScopeType } from "@fluidframework/protocol-definitions";
+import { BasicRestWrapper } from "@fluidframework/server-services-client";
+import { IContainerManager, IDocument } from "@fluidframework/server-services-core";
+import { generateToken, getCorrelationId } from "@fluidframework/server-services-utils";
+
+/**
+ * Manager to fetch container from Alfred using the internal URL.
+ */
+export class ContainerManager implements IContainerManager {
+
+	constructor(
+		private readonly internalAlfredUrl: string,
+		private readonly key: string
+	) {}
+
+	public async readContainer(tenantId: string, containerId: string): Promise<IDocument> {
+		const restWrapper = await this.getBasicRestWrapper(tenantId, containerId);
+		const resultP = restWrapper.get<IDocument>(
+			`/documents/${tenantId}/${containerId}`,
+		);
+		return resultP;
+	}
+
+	private async getBasicRestWrapper(tenantId: string, containerId: string) {
+		const getDefaultHeaders = () => {
+			const token = { jwt: generateToken(tenantId, containerId, this.key, [ScopeType.DocRead]) };
+			return {
+				Authorization: `Basic ${token.jwt}`,
```

Comment:
nit, the `token` object wrapper is unnecessary. You could just use
```ts
const jwt = generateToken(...);
return {
    Authorization: `Basic ${jwt}`,
}
```

Reply:
Addressed